### ISSUE TITLE
Internal PDF Viewer hanging

### ISSUE THREAD

**Issue by eric-silk** [James-Yu/LaTeX-Workshop#3825]
> _Please fill the following fields with a star (*) and provide as much related information as possible._

## Pre-checks*
> _Please change the following `[ ]` to `[x]` for confirmation._
- [x] The issue has not been reported in this repository.
- [x] The issue remains after disabling all other extensions and restarting Visual Studio Code.
- [x] The [FAQ](https://github.com/James-Yu/LaTeX-Workshop/wiki/FAQ) cannot address the issue.
- [x] The issue is not related to compiling a document, or the document can be successfully compiled in the OS terminal but not in Visual Studio Code with this extension.

## Environment*
> _Please write exact version numbers instead of descriptors such as `latest`._
- Operating System: [e.g. Windows 10] Kubuntu 22.04.2
- Visual Studio Code Version: [e.g. 1.41.0] 1.77.0
- LaTeX Workshop Version: [e.g. 8.5.0] v9.8.2
- TeX Distribution Version: [e.g. TeX Live 2019 / MiKTeX 2.9.7250] TeX Live 2022

> _Please list the environment and version number if you are using VSCodium, Snap or Flatpack versions of Visual Studio Code, and/or Visual Studio Code Remote Containers/SSH/WSL._
- N/A

## The Issue*
> _Please briefly describe the issue you come across._
- Upon attempting to open the internal PDF viewer, a new tab opens but is left blank. After some time (on the order of minutes) a dialogue appears indicating "The window is not responding" with the ability to close, keep waiting, or re-open. Any but "keep waiting" cause VS Code to close without reopening. If "keep waiting" is selected and the tab is not closed, this dialogue reappears as before. If the tab is exited manually, it will not.
- I have completely removed all plugins and uninstalled code via `apt purge code`, rebooted, and reinstalled and the issue persists.
- Inspecting the logs, if I open the PDF URL listed it opens in my browser and renders correctly.
- This issue occurs even when the `--disable-gpu` arg is provided upon launch.

### Reproduction Steps
> _Please list out the steps to reproduce your bug. Include relevant environmental variables or any other configuration._
1. Compile a TeX document, either through the extension or not.
2. Attempt to open the internal PDF viewer via the "View LaTeX PDF File" button, or by opening the PDF through VS Code's file explorer.

### Expected Behavior
> _What were you expecting to see? Include any relevant examples or documentation links._
- I'd expect the PDF to open and be viewable.

### LaTeX Workshop Output*
> _Please paste the whole log messages below, not parts of ones. The log should start with `New log placeholder %WS1% registered`._
```
[Paste the log here. Do not remove the surrounding backquotes (`).]
```

### Developer Tools Console
> _Please paste the whole log messages below, not parts of ones. This console logs can sometimes be very important in many cases. To access the log, click `Help` -> `Toggle Developer Tools` -> `Console`._
```
[16:46:55][Logger] New log placeholder %WS1% registered for /home/eric/Documents/PhD/ECE528 .
[16:46:55][Config] latex-workshop.latex.recipes: [{"name":"latexmk","tools":["latexmk"]},{"name":"pwd","tools":["pwd"]}] .
[16:46:55][Config] latex-workshop.latex.tools: [{"name":"pwd","command":"pwd","args":[],"env":{}},{"name":"latexmk","command":"latexmk","args":["-synctex=1","-interaction=nonstopmode","-file-line-error","-pdf","-outdir=%OUTDIR%","%DOC%"],"env":{}},{"name":"lualatexmk","command":"latexmk","args":["-synctex=1","-interaction=nonstopmode","-file-line-error","-lualatex","-outdir=%OUTDIR%","%DOC%"],"env":{}},{"name":"xelatexmk","command":"latexmk","args":["-synctex=1","-interaction=nonstopmode","-file-line-error","-xelatex","-outdir=%OUTDIR%","%DOC%"],"env":{}},{"name":"latexmk_rconly","command":"latexmk","args":["%DOC%"],"env":{}},{"name":"pdflatex","command":"pdflatex","args":["-synctex=1","-interaction=nonstopmode","-file-line-error","%DOC%"],"env":{}},{"name":"bibtex","command":"bibtex","args":["%DOCFILE%"],"env":{}},{"name":"rnw2tex","command":"Rscript","args":["-e","knitr::opts_knit$set(concordance = TRUE); knitr::knit('%DOCFILE_EXT%')"],"env":{}},{"name":"jnw2tex","command":"julia","args":["-e","using Weave; weave(\"%DOC_EXT%\", doctype=\"tex\")"],"env":{}},{"name":"jnw2texmintex","command":"julia","args":["-e","using Weave; weave(\"%DOC_EXT%\", doctype=\"texminted\")"],"env":{}},{"name":"tectonic","command":"tectonic","args":["--synctex","--keep-logs","%DOC%.tex"],"env":{}}] .
[16:46:55][Config] editor.acceptSuggestionOnEnter: "off" .
[16:46:55][Manager] Set $LATEXWORKSHOP_DOCKER_LATEX: ""
[16:46:55][Server] Creating LaTeX Workshop http and websocket server.
[16:46:55][Server] Server successfully started: {"address":"127.0.0.1","family":"IPv4","port":45081} .
[16:46:55][Extension] Initializing LaTeX Workshop.
[16:46:55][Extension] Extension root: /home/eric/.vscode/extensions/james-yu.latex-workshop-9.8.2
[16:46:55][Extension] $PATH: /home/eric/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
[16:46:55][Extension] $SHELL: /bin/bash
[16:46:55][Extension] $LANG: en_US.UTF-8
[16:46:55][Extension] $LC_ALL: undefined
[16:46:55][Extension] process.platform: linux
[16:46:55][Extension] process.arch: x64
[16:46:55][Extension] vscode.env.appName: Visual Studio Code
[16:46:55][Extension] vscode.env.remoteName: undefined
[16:46:55][Extension] vscode.env.uiKind: 1
[16:46:55][Extension] LaTeX Workshop initialized.
[16:46:55][Format][Bib] Bibtex format config: {"tab":"  ","case":"lowercase","left":"{","right":"}","trailingComma":false,"sort":["key"],"alignOnEqual":true,"sortFields":false,"fieldsOrder":[],"firstEntries":["string","xdata"]}
[16:46:55][Extension] Trigger characters for intellisense of LaTeX documents: ["\\",",","{"]
[16:46:55][Manager] Current workspace folders: ["file://%WS1%"]
[16:46:55][Manager] Found root file from active editor: %WS1%/hw4/hw4.tex
[16:46:55][Manager] Root file changed: from undefined to %WS1%/hw4/hw4.tex
[16:46:55][Manager] Start to find all dependencies.
[16:46:55][Manager] Root file languageId: latex
[16:46:55][Cacher][Watcher] Reset.
[16:46:55][Cacher] Adding %WS1%/hw4/hw4.tex .
[16:46:55][Cacher][Watcher] Watched %WS1%/hw4/hw4.tex with a new watcher on %WS1%/hw4 .
[16:46:55][Cacher] Caching %WS1%/hw4/hw4.tex .
[16:46:55][Cacher] Updated inputs of %WS1%/hw4/hw4.tex .
[16:46:55][DupLabel] Checking for duplicate labels: %WS1%/hw4/hw4.tex .
[16:46:55][Cacher] Updated elements of %WS1%/hw4/hw4.tex .
[16:46:55][Cacher] Updated bibs of %WS1%/hw4/hw4.tex .
[16:46:55][Cacher] Cached %WS1%/hw4/hw4.tex .
[16:46:55][Cacher] Parsing .fls %WS1%/hw4/hw4.fls .
[16:46:55][Cacher] Found .aux %WS1%/hw4/hw4.tex from .fls %WS1%/hw4/hw4.fls , parsing.
[16:46:55][Cacher] Parsed .aux %WS1%/hw4/hw4.tex .
[16:46:55][Cacher] Parsed .fls %WS1%/hw4/hw4.fls .
[16:46:55][Structure] Structure force updated with 5 root sections for %WS1%/hw4/hw4.tex .
[16:46:55][Server] valdOrigin is http://127.0.0.1:45081
[16:47:05][Viewer][Panel] Internal PDF viewer at http://127.0.0.1:45081/viewer.html?file=pdf..ZmlsZSUzQSUyRiUyRiUyRmhvbWUlMkZlcmljJTJGRG9jdW1lbnRzJTJGUGhEJTJGRUNFNTI4JTJGaHc0JTJGaHc0LnBkZg .
[16:47:05][Cacher][Watcher] Watched %WS1%/hw4/hw4.pdf with a new watcher on %WS1%/hw4 .
[16:47:05][Viewer] Open PDF tab for file://%WS1%/hw4/hw4.pdf
```

## Anything Else?
> _Add any other context about the problem below._
![image](https://user-images.githubusercontent.com/17257092/229931034-43a00ad8-a1bf-4a8b-b984-6e462ded6692.png)

**Comment by eric-silk** [comment]
Other interesting behavior...attempting to open Google Chrome at all after this occurs will result in the same behavior -- even the built-in pages fail to load (e.g. chrome://welcome, etc.) and report a "Page Unresponsive" with options to exit or wait. Launching chrome via the terminal produces the same behavior, UNLESS I also pass `google-chrome --enable-logging --v=1`.

I'm not super familiar with how VSCode works, but could this actually be a problem with Electron/Chromium?

**Comment by eric-silk** [comment]
Apologies for the frequent posts but I have found a temporary fix: running code with `--verbose` appears to resolve the issue. Most curiously, after running it this way once, a subsequent run without the argument also appears to work as expected. If this doesn't happen again in the near future, I'll close the issue as unable to replicate. Otherwise, I'll update if I can more precisely describe the conditions that cause it.

Similarly, Chrome appears to require the `--v=1` argument (not the `--enable-logging` arg) but does _not_ have the fix persist after closing and restarting without the argument.

**Comment by James-Yu** [comment]
It is most likely a PDF problem. Do you observe the same behavior with a minimal LaTeX-generated PDF?

**Comment by eric-silk** [comment]
No, this was the case with seemingly any PDF.

**Comment by James-Yu** [comment]
> No, this was the case with seemingly any PDF.

If so, it is almost for sure a local problem. Unfortunately I do not have any clue on a possible cause.

**Comment by eric-silk** [comment]
I believe its an issue with Chromium, I've opened a bug on their tracker.

**Comment by alanlivio** [comment]
Hi both @James-Yu and @eric-silk. I have the same problem using WSL.

LaTeX Workshop v9.8.2
code  1.77.1

```
4:46:19][Builder] Successfully built %WS1%/main.tex .
[14:46:21][Commander] BUILD command invoked.
[14:46:21][Commander] The document of the active editor: file://%WS1%/main.tex
[14:46:21][Commander] The languageId of the document: latex
[14:46:21][Manager] Current workspace folders: ["file://%WS1%"]
[14:46:21][Manager] Found root file from active editor: %WS1%/main.tex
[14:46:21][Manager] Keep using the same root file: %WS1%/main.tex
[14:46:21][Structure] Structure updated with 13 root sections for %WS1%/main.tex .
[14:46:21][Commander] Building root file: %WS1%/main.tex
[14:46:21][Builder] Build root file %WS1%/main.tex
[14:46:21][Builder] outDir: %WS1% .
[14:46:21][Builder] Preparing to run recipe: latexmk.
[14:46:21][Builder] Prepared 1 tools.
[14:46:21][Builder] Recipe step 1 The command is latexmk:["-synctex=1","-interaction=nonstopmode","--shell-escape","-file-line-error","-pdf","-outdir=%WS1%","%WS1%/main"].
[14:46:21][Builder] env: {}
[14:46:21][Builder] root: %WS1%/main.tex
[14:46:21][Builder] cwd: %WS1%
[14:46:21][Builder] LaTeX build process spawned with PID 12963.
[14:46:21][Builder] Finished a step in recipe with PID 12963.
[14:46:21][Builder] Successfully built %WS1%/main.tex .
[14:46:21][Commander] VIEW command invoked with mode: file://%WS1%/main.tex.
[14:46:21][Manager] Current workspace folders: ["file://%WS1%"]
[14:46:21][Manager] Found root file from active editor: %WS1%/main.tex
[14:46:21][Manager] Keep using the same root file: %WS1%/main.tex
[14:46:21][Structure] Structure updated with 13 root sections for %WS1%/main.tex .
[14:46:21][Viewer][Panel] Internal PDF viewer at http://127.0.0.1:39135/viewer.html?file=pdf..ZmlsZSUzQSUyRiUyRiUyRmhvbWUlMkZhbGFuJTJGc3JjJTJGZHJhZnQtZnVuZGluZyUyRm1haW4ucGRm .
[14:46:21][Cacher][Watcher] Watched %WS1%/main.pdf .
[14:46:22][Viewer] Open PDF tab for file://%WS1%/main.pdf
[14:46:26][Structure] Structure updated with 13 root sections for %WS1%/main.tex .
```

**Comment by eric-silk** [comment]
Is the chromium instance being launched for the PDF viewer in the host environment (Windows) or in WSL?

**Comment by CastleStar14654** [comment]
@eric-silk for Remote-WSL, LaTeXworkshop is launched in WSL (and so does the PDF viewer). Your problem stated in the issue is on a full Kubuntu instead?

**Comment by CastleStar14654** [comment]
Now, wierd things happend: I tried using the external PDF viewer instead, and then, when I relaunched the internal tab viewer, it works... Hope this could help you.

**Comment by alanlivio** [comment]
Hi @CastleStar14654, @eric-silk , @syvshc.
I created [this discussion ](https://github.com/James-Yu/LaTeX-Workshop/discussions/3833), and @nidrissi posted that he resolved it by using WSL 1.2.1. It worked for me as well. If any wants to try, I get it from [here ](https://github.com/microsoft/WSL/releases/tag/1.2.1).
Best,
